frontend/src/views/RoleView.vue

```diff
@@ -1,29 +1,76 @@
-<script lang="ts" setup>
-import { ref } from 'vue'
-import { ElButton, ElDialog, ElInput } from 'element-plus'
-import { useKnowledgeStore } from '@/stores/knowledges/index'
+<template>
+  <div class="PageWrapper">
+    <div class="TableContainer">
+      <h1>{{ role?.name }}</h1>
+      <h2>Attached Knowledges</h2>
+      <br />
+      <ElTable :data="knowledges" style="width: 100%">
+        <ElTableColumn prop="name" label="Knowledges" />
+        <ElTableColumn fixed="right" label="Actions" width="150">
+          <template #default="{ row }">
+            <div>
+              <ElButton @click="onDeleteRow(row)" type="text" size="small">Delete</ElButton>
+              <ElButton @click="onEditRow(row)" type="text" size="small">Edit</ElButton>
+            </div>
+          </template>
+        </ElTableColumn>
+      </ElTable>
+      <div class="demo">
+        <br />
+        <ElButton type="primary" @click="showDialogAddNewKnowledge">Add new Knowledge</ElButton>
+        <ElDialog v-model="createKnowledgeDialogFormVisible" title="New Knowledge" width="500">
+          <ElInput v-model="newKnowledgeName" placeholder="Knowledge name" :clearable="false" />
+          <template #footer>
+            <div class="dialog-footer">
+              <ElButton @click="createKnowledgeDialogFormVisible = false">Cancel</ElButton>
```

Comment:
```suggestion
              <ElButton @click="hideDialog">Cancel</ElButton>
```
You already have a method that does this